File: frameworks/java/spring-boot/Implementation/src/main/java/implementation/routes/AuthorizedRoutes.java
package implementation.routes;

import implementation.Payload;
import implementation.Payloads;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/** authorized: the route Spring Security guards, configured in AuthorizedSecurity. */
@RestController
public class AuthorizedRoutes {

    private final Payloads p;

    AuthorizedRoutes(Payloads p) {
        this.p = p;
    }

    @GetMapping("/authorized/small")
    public Payload small() {
        return p.small();
    }
}

File: frameworks/dotnet/hardened/Implementation/Routes/AuthorizedRoutes.cs
using Hardened.Requests.Runtime.Authorization;
using Hardened.Web.Runtime.Attributes;

namespace Implementation.Routes;

/// <summary>
/// authorized: Hardened's authorization, which checks the caller's grants before the request is
/// bound. BearerTokenSource reads the token into the caller on every request.
/// </summary>
public class AuthorizedRoutes(Payloads payloads)
{
    [Get("/authorized/small")]
    [AuthorizeGrants(Grants.ReadSmall)]
    public Payload Small() => payloads.Small;
}
